src/org/zaproxy/zap/extension/soap/WSDLCustomParser.java

```diff
@@ -489,6 +491,16 @@ private static void persistMessage(final HttpMessage message) {
 				@Override
 				public void run() {
 					extHistory.addHistory(historyRef);
+					/* Modifies the URI adding the SOAP operation name to avoid overwrites. It's done after
+					 * saving it in history so that the original URI is preserved for scanning tasks.*/
+					try {
+						URI soapURI = message.getRequestHeader().getURI();
+						String soapStringURI = soapURI.getPath();
+						soapURI.setPath(soapStringURI+opDisplayName);
+						message.getRequestHeader().setURI(soapURI);
```

Comment:
Since only the path is being changed this statement is not necessary (the method ```HttpRequestHeader.getURI()``` does not return a copy so the change done to the path will be reflected when URI is used internally).